inc/saitam.h

```diff
@@ -1239,7 +1277,7 @@ typedef enum _sai_tam_report_attr_t
     SAI_TAM_REPORT_ATTR_REPORT_MODE,
 
     /**
-     * @brief Report Interval in micro seconds
```

Comment:
why remove unit ?

Reply:
The new units attribute means the units will not necessarily be microseconds, though they are microseconds by default, for backward-compatibility. The user may change it, e.g. to nanoseconds (usually hundreds of nanoseconds) for a specific use-case.

Reply:
woot ? how come parameter can chance type of units ? this seems like a wrong idea, please elaborate more

Reply:
Hi, this is modelled after: SAI_TAM_TELEMETRY_ATTR_TAM_REPORTING_UNIT and SAI_TAM_TELEMETRY_ATTR_REPORTING_INTERVAL. That one uses seconds/minutes/days.

The requirement is to support sub-microsecond interval for this attribute. By defaulting to microsecond units, but allowing the user to change the units to nanoseconds, it can allow configuring for reduced interval while maintaining backward compatibility.

Reply:
@kcudnik Is it okay? If not, any suggestion? Just simply changing the units to be in nanoseconds would be the other approach but seems like a landmine if a user or implementation misses the update.

Reply:
thanks i get it now